### PR TITLE
#192 - Project documentation

### LINKED ISSUE #81: Configurable college information outside vue templates
It would be nice if common data for school used through the site was configurable through .csv file. It could contain school's name, path to logo, link to the main site, maybe something else...

We also need to consider generic website's descriptions. They cannot be only for college/university, but should be easily adjustable. Some high schools require internship and they might also like to use this template.

~~For the same reason I'm not the fan of the second section ['Join us!'] on the Landing page. It sounds like we are the ones managing the site, and schools can apply to have a 'section' for themselves. The project is intended to give them full administration rights to the whole thing.~~ As of #163

### LINKED ISSUE #192: Create a documentation page
The project aims to allow colleges to create a list of companies looking for interns.

For that we should create a documentation to explain features and structure of the application.

It should probably have two parts, like:
- Technical:
  - General description of the app and overview of technology used
  - Project structure
  - Setup and installation
  - How to run seeders and test application manually (check webpage, see database for example in DataGrip, tinker)
  - How to run automatic tests (when they get implemented)
  - How to create an admin account
  - Intended behaviour and implementations / other stuff I did not think of right now
- End user:
  - Nice overview of what this application is with fancy screenshots
  - How to use the company browser
  - How to create a company, explanation of the verification process
  - Admin panel and all related stuff
  - And more?
 
I think the documentation page could be hosted in Github Pages as a static website, maybe using a [VuePress](https://vuepress.vuejs.org/) template?

### PR DESCRIPTION
**Pull request by sbanasiak**
Added documentation of the project - technical and end user.
- **Technical documentation** contains information about the application, configuration and setup instructions. 
- **User documentation** contains instructions for using specific functionalities of the application.

imqcqeN

1. The documentation is located in the `docs` folder and is independent of the rest of the project.
2. Command to run the documentation:
`docker-compose exec node-docs npm run dev`
It should appear at: http://localhost:8080/
3. A new action has been added that builds documentation and publishes at:
https://blumilksoftware.github.io/internships/ (should be available after pushing on main)
For the correct behavior of the action, there is a need to switch the repository `Build and deployment` Source setting under the `Pages` tab to `GitHub Actions (Beta)`:
chrome_0zIruwsGn7
4. Along with the documentation, a `assets.json` file has been added to the `frontend` folder of the project, which allows to configure basic data about the university.
It looks like this:
```json
{
    "college": {
        "name": "Collegium Witelona Uczelnia Państwowa",
        "website": "http://www.pwsz.legnica.edu.pl/",
        "latitude": 51.2048546,
        "longitude": 16.1472681
    }
}
```
Closes #192 
Closes #81

### CODE REVIEW

File: docs/pl/technical/commands.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+```
+docker-compose exec {serwis} {komenda}
+```
+
+W miejscu {serwis} należy wpisać odpowiednią nazwę serwisu (wszystkie serwisy znajdują się w pliku docker-compose.yml):
+@[code](@/docker/service-list.md)
+
+::: tip
+Więcej informacji odnośnie polecenia exec znajduje się w dokumentacji Dockera: [https://docs.docker.com/engine/reference/commandline/compose_exec](https://docs.docker.com/engine/reference/commandline/compose_exec/)
```
Comment (EwelinaLasowy) [inline docs/pl/technical/commands.md:15]:
```suggestion
Więcej informacji odnośnie polecenia `exec` znajduje się w dokumentacji Dockera: [https://docs.docker.com/engine/reference/commandline/compose_exec](https://docs.docker.com/engine/reference/commandline/compose_exec/)
```

File: docs/pl/technical/commands.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+:::
+
+## Tworzenie konta administratora
+Tworzenie konta administratora odbywa się za pomocą polecenia artisan z żądanym e-mailem, hasłem, imieniem i nazwiskiem:
+```
+php artisan make:admin {email} {hasło} {imię} {nazwisko}
+```
+
+Przykładowe użycie komendy:
```
Comment (EwelinaLasowy) [inline docs/pl/technical/commands.md:24]:
```suggestion
Przykładowe użycie polecenia:
```

File: docs/pl/technical/commands.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+```
+php artisan make:admin email@example.com haslo123 Jan Kowalski
+```
+
+## Lokalizowanie komendą locate:place
```
Comment (EwelinaLasowy) [inline docs/pl/technical/commands.md:29]:
```suggestion
## Lokalizowanie poleceniem locate:place
```

File: docs/pl/technical/commands.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+W celu przetestowania serwisu od geokodowania, można skorzystać z komendy:
```
Comment (EwelinaLasowy) [inline docs/pl/technical/commands.md:30]:
```suggestion
W celu przetestowania serwisu od geokodowania, można skorzystać z polecenia:
```

File: docs/pl/technical/migrations-and-seeders.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# Migracje i seedery
+Do prawidłowego funkcjonowania aplikacji, baza danych musi mieć odpowiednią strukturę. W tym celu zostały stworzone odpowiednie migracje oraz seedery.
+
+## Migracje
+Migracje tworzą jedynie strukturę bazy danych, ale nie wypełniają jej danymi. W celu wywołania migracji, należy użyć komendy:
```
Comment (EwelinaLasowy) [inline docs/pl/technical/migrations-and-seeders.md:5]:
```suggestion
Migracje tworzą jedynie strukturę bazy danych, ale nie wypełniają jej danymi. W celu wywołania migracji, należy użyć polecenia:
```

File: docs/pl/technical/migrations-and-seeders.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+```
+php artisan migrate
+```
+
+Jeżeli z jakiegoś powodu potrzebujemy przebudować całą bazę danych (wyczyścic i stworzyć strukturę na nowo), istnieje polecenie:
```
Comment (EwelinaLasowy) [inline docs/pl/technical/migrations-and-seeders.md:10]:
```suggestion
Jeżeli z jakiegoś powodu potrzebujemy przebudować całą bazę danych (wyczyścić i stworzyć strukturę na nowo), istnieje polecenie:
```

File: docs/pl/technical/migrations-and-seeders.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+::: tip
+Więcej informacji odnośnie migracji znajduje się w [dokumentacji Laravela](https://laravel.com/docs/9.x/migrations)
+:::
+
+## Seedery
+Seedery wypełniają danymi utworzoną strukturę bazy danych. W aplikacji zostały utworzone dwa główne seedery: struktury uczelni oraz losowych danych.
+
+### Seeder struktury uczelni
+Seeder struktury uczelni uzupełnia bazę danych informacjami na temat struktury uczelni: nazwami wydziałów, kierunkami oraz specjalizacjami. Są to dane potrzebne do prawidłowego dodawania oraz filtrowania firm. Uruchomienie seedera struktury uczelni następuje przez wywołanie komendy:
```
Comment (EwelinaLasowy) [inline docs/pl/technical/migrations-and-seeders.md:23]:
```suggestion
Seeder struktury uczelni uzupełnia bazę danych informacjami na temat struktury uczelni: nazwami wydziałów, kierunkami oraz specjalizacjami. Są to dane potrzebne do prawidłowego dodawania oraz filtrowania firm. Uruchomienie seedera struktury uczelni następuje przez wywołanie polecenia:
```

File: docs/pl/technical/run.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
+# Uruchomienie
+
+## Środowisko developerskie przy użyciu Dockera
+Aplikacja została skonfigurowana pod uruchomienie w skonteneryzowanym środowisku za pomocą oprogramowania Docker i narzędzia Docker Compose. Dzięki czemu, można ją uruchomić na dowolnym środowisku, niezależnie od jego rodzaju i konfiguracji infrastruktury.
```
Comment (EwelinaLasowy) [inline docs/pl/technical/run.md:4]:
```suggestion
Aplikacja została skonfigurowana pod uruchomienie w skonteneryzowanym środowisku za pomocą oprogramowania Docker i narzędzia Docker Compose. Dzięki temu można ją uruchomić na dowolnym środowisku, niezależnie od jego rodzaju i konfiguracji infrastruktury.
```

File: docs/pl/user/index.md [hunk 9]
```diff
@@ -0,0 +1,4 @@
+# Wprowadzenie
+
+## Dokumentacja techniczna
+Jeżeli jesteś zainteresowany konfiguracją oraz uruchomieniem aplikacji na własnym urządzeniu, przejdź na [Dokumentacja techniczna](../technical)
```
Comment (EwelinaLasowy) [inline docs/pl/user/index.md:4]:
```suggestion
Jeżeli chcesz skonfigurować oraz uruchomić aplikację na własnym urządzeniu, przejdź na [Dokumentacja techniczna](../technical)
```

File: docs/pl/user/manage-company.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+W celu ustawienia poprawnego położenia firmy po uzupełnieniu danych adresowych należy kliknąć w przycisk `Ustaw marker`. Jeśli położenie firmy jest niepoprawne, można dostosować marker przez przeciągniecie.
+
+![Przesuwanie markera](./images/company-map.gif)
+
+Edycja opisu firmy odbywa się przez specjalny edytor wykorzystujący język Markdown. Możesz korzystać z przycisków znajdujących się w menu lub samemu napisać odpowiednie znaczniki.
+
+![Edytor opisu](./images/company-markdown.png)
+
+## Przeglądanie własnych firm
+Lista dodanych przez ciebie firm znajduje się pod przyciskiem `Zarządzaj firmami` w pasku nawigacyjnym.
```
Comment (EwelinaLasowy) [inline docs/pl/user/manage-company.md:17]:
```suggestion
Lista dodanych przez Ciebie firm znajduje się pod przyciskiem `Zarządzaj firmami` w pasku nawigacyjnym. 
```

File: docs/pl/user/manage-company.md [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+## Przeglądanie własnych firm
+Lista dodanych przez ciebie firm znajduje się pod przyciskiem `Zarządzaj firmami` w pasku nawigacyjnym. 
+
+![Lista własnych firm](./images/company-list.png)
+Firma, która jeszcze nie została zweryfikowana przez moderatora strony będzie miała status `W oczekiwaniu na weryfikację`.
+
+Po kliknięciu w kafelek z firmą można zobaczyć jej opis oraz ją [usunąć](#usuwanie-firmy).
+## Usuwanie firmy
+W widoku z opisem Twojej firmy możesz ją usunąc. W tym celu odszukaj czerwony przycisk na dole strony z ikoną kosza na śmieci.
```
Comment (EwelinaLasowy) [inline docs/pl/user/manage-company.md:24]:
```suggestion
W widoku z opisem Twojej firmy możesz ją usunąć. W tym celu odszukaj czerwony przycisk na dole strony z ikoną kosza na śmieci.
```

File: docs/pl/user/manage-company.md [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+![Usuwanie firmy](./images/company-delete.png)
+Przed usunięciem firmy zostanie wyświetlone stosowny komunikat potwierdzający zapytanie.
```
Comment (EwelinaLasowy) [inline docs/pl/user/manage-company.md:27]:
```suggestion
Przed usunięciem firmy zostanie wyświetlone stosowny komunikat potwierdzający usunięcie.
```

File: docs/pl/user/manage-site.md [hunk 13]
```diff
@@ -0,0 +1,9 @@
+# Zarządzanie stroną
```
Comment (EwelinaLasowy) [inline docs/pl/user/manage-site.md:1]: Why here is empty file?

File: frontend/assets.json [hunk 14]
```diff
@@ -0,0 +1,8 @@
+{
+    "college": {
+        "name": "Collegium Witelona Uczelnia Państwowa",
+        "website": "http://www.pwsz.legnica.edu.pl/",
+        "latitude": 51.2048546,
+        "longitude": 16.1472681
+    }
+}
```
Comment (EwelinaLasowy) [inline frontend/assets.json:8]:
```suggestion
}

```

File: frontend/assets/lang/en.json [hunk 15]
```diff
@@ -181,4 +182,4 @@
         "letterCount": "Letter count",
         "scrollAuto": "Scroll auto"
     }
-}
+}
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/en.json:185]:
```suggestion
}

```

File: frontend/assets/lang/pl.json [hunk 16]
```diff
@@ -182,4 +183,4 @@
         "letterCount": "Liczba znaków",
         "scrollAuto": "Scrollowanie automatyczne"
     }
-}
+}
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/pl.json:186]:
```suggestion
}

```

File: frontend/js/Pages/Company/Components/MapDisplay.vue [hunk 17]
```diff
@@ -156,8 +151,7 @@ onMounted(() => {
     style: `https://api.maptiler.com/maps/streets/style.json?key=${
       import.meta.env.VITE_MAPLIBRE_TOKEN
     }`,
-    center: [16.1472681, 51.2048546],
-    zoom: 15,
+    center: [assets.college.longitude, assets.college.latitude],    zoom: 15,
```
Comment (EwelinaLasowy) [inline frontend/js/Pages/Company/Components/MapDisplay.vue:154]:
```suggestion
    center: [assets.college.longitude, assets.college.latitude],
    zoom: 15,
```

**Review by EwelinaLasowy — COMMENTED** [review]
1. I can't generate a documentation:
image

(I killed all docker containers and run `sh setup.sh`, but it doesn't help)
2. Please look more one time at this documentation and mark all commands and file names` like this `. Somewhere I marked it for you, but not in all places.
3. Why in Polish documentation (I mean in `docs/pl/user/images/`) we have pictures in English? For example account-reset.png, `account-verify.png`.  EDIT: Oh, we don't translate e-mails - what a pity :(

File: docs/en/technical/commands.md [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+If the application is running in Docker containers (for example, it was launched using [convenient script](../technical/run.html#using-a-convenient-script)), in order to send commands, use docker-compose in the root directory of the application:
+
+```
+docker-compose exec {service} {command}
+```
+
+In the {service}, enter the appropriate service name (all services are in the docker-compose.yml file):
+@[code](@/docker/service-list.md)
+
+::: tip
```
[the remark this answers is not in the document]
Comment (sbanasiak) [reply docs/en/technical/commands.md:14]: Yes, it generates tips as below:
image

**Comment by sbanasiak** [comment]
> 1. I can't generate a documentation:
>    image
> 
> (I killed all docker containers and run `sh setup.sh`, but it doesn't help) 
> 2. Please look more one time at this documentation and mark all commands and file names`like this`. Somewhere I marked it for you, but not in all places. 
> 3. Why in Polish documentation (I mean in `docs/pl/user/images/`) we have pictures in English? For example account-reset.png, `account-verify.png`. EDIT: Oh, we don't translate e-mails - what a pity :(

1. You need to install required dependencies: `docker-compose exec node-docs npm install`
2. I denoted other names, commands as `code`.
3. Yes, we still need to figure it out

I also removed empty files and accepted the other suggestions.

**Review by EwelinaLasowy — APPROVED** [review]

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**